extensions/amp-ad/0.1/amp-ad.js

```diff
@@ -69,11 +69,28 @@ export class AmpAd extends AMP.BaseElement {
 
       this.win.ampAdSlotIdCounter = this.win.ampAdSlotIdCounter || 0;
       const slotId = this.win.ampAdSlotIdCounter++;
+      let slotIdByUnit;
+
+      if (this.element.hasAttribute('data-slot')) {
+        const dataSlot = this.element.getAttribute('data-slot');
+        this.win.ampAdSlotIdByUnitCounter =
+          this.win.ampAdSlotIdByUnitCounter || {};
+        this.win.ampAdSlotIdByUnitCounter[dataSlot] =
+          this.win.ampAdSlotIdByUnitCounter[dataSlot] || 0;
+        slotIdByUnit = this.win.ampAdSlotIdByUnitCounter[dataSlot]++;
+      }
```

Comment:
```suggestion
      this.win.ampAdUnitSlotIdCounter =
        this.win.ampAdUnitSlotIdCounter || {};
      const dataSlot = this.element.getAttribute('data-slot');
      this.win.ampAdUnitSlotIdCounter[dataSlot] =
        this.win.ampAdUnitSlotIdCounter[dataSlot] || 0;
      const unitSlotId = this.win.ampAdUnitSlotIdCounter[dataSlot]++;
```

Reply:
@powerivq I believe that if remove the `if` who check if the element has the attribute `data-slot` it could return `null`  to the `dataSlot` variable and probably would try to set a property in the object with key `null`

Reply:
Yeah, but there are no problems with setting a key `null` or `undefined`